### ISSUE TITLE
Wishlist: xterm 256 color support

### ISSUE THREAD

**Issue by perlun** [nicoulaj/rainbow#67]
Hi,

Thanks for a nice tool! :+1: I discovered it today since I wanted to get better outputs from tailing my Tomcat & application logs.

One thing I noted that seems to be missing from the feature matrix which would be incredibly nice is to be able to use [8-bit colors](https://en.wikipedia.org/wiki/ANSI_escape_code#8-bit). They are supported by some of the modern terminal emulators - very nice since it gives a much larger amount of colors to select from.

The Ruby library which funnily is your namesake https://github.com/sickill/rainbow has this. Its implementation is particularly interesting, since it allows you to enter an RGB value (say `#FF00FF`) which will then be automatically converted to the closest 8-bit color code.

That makes the tool incredibly nice to use, but is obviously a bit more work. Just being able to specify these extended ANSI escape codes would be a good start in this area.

**Comment by nicoulaj** [comment]
Indeed, and some terminals even support true color.

The difficulty here is more to define a usable config syntax. One intermediate step I could see is to introduce support for color themes, so that user can replace the default 8 colors (black, white, red, green, yellow, blue, magenta, cyan) with 8/24 bit ones. But that would still be limited to 8 colors.

Related: #2

**Comment by perlun** [comment]
> The difficulty here is more to define a usable config syntax

I agree, right now the syntax is pretty much connected to the ANSI 3/4-bit colors so this is a bit of a change from that perspective.

Maybe something like this would do:

```
[filters]
faint: ^[A-Z][a-z]+\s\d+,\s\d+\s\d+:\d+:\d+|^\d+\s[a-z]+\s\d+\s\d+:\d+:\d+
FF0000: ^GRAVE:\s.*|^SEVERE:\s.*
FFFF00: ^ATTENTION:\s.*|^WARNING:\s.*
```

(Can't really prefix them with `#` since we use that for commenting. 😊)
